File: app/src/test/java/com/permissionnanny/missioncontrol/AppControlActivityTest.java
package com.permissionnanny.missioncontrol;

import android.view.Menu;
import android.view.MenuItem;
import com.permissionnanny.NannyAppTestCase;
import com.permissionnanny.dagger.MockActivityComponent;
import com.permissionnanny.dagger.MockComponentFactory;
import javax.inject.Inject;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.robolectric.Robolectric;
import org.robolectric.util.ActivityController;

import static org.mockito.Mockito.verify;

public class AppControlActivityTest extends NannyAppTestCase {

    ActivityController<AppControlActivity> mController;
    AppControlActivity mAppControlActivity;
    @Inject AppControlBinder mAppControlBinder;
    @Mock Menu mMenu;
    @Mock MenuItem mMenuItem;

    @Before
    public void setUp() throws Exception {
        mController = Robolectric.buildActivity(AppControlActivity.class);
        mAppControlActivity = mController.get();
        MockActivityComponent contextComponent = MockComponentFactory.getActivityComponent();
        mAppControlActivity.inject(contextComponent);
        contextComponent.inject(this);
        mController.setup();
    }

    @Test
    public void lifecycle() throws Exception {
        mController.pause().stop().destroy();
        verify(mAppControlBinder).onCreate(null);
        verify(mAppControlBinder).onResume();
        verify(mAppControlBinder).onPause();
    }

    @Test
    public void onCreateOptionsMenu() throws Exception {
        mAppControlActivity.onCreateOptionsMenu(mMenu);

        verify(mAppControlBinder).onCreateOptionsMenu(mMenu);
    }

    @Test
    public void onOptionsItemSelected() throws Exception {
        mAppControlActivity.onOptionsItemSelected(mMenuItem);

        verify(mAppControlBinder).onOptionsItemSelected(mMenuItem);
    }
}

File: app/src/test/kotlin/com/permissionnanny/missioncontrol/AppControlActivityTest.kt
package com.permissionnanny.missioncontrol

import android.view.Menu
import android.view.MenuItem
import com.permissionnanny.NannyAppTestCase
import com.permissionnanny.dagger.MockComponentFactory
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.robolectric.Robolectric
import org.robolectric.util.ActivityController
import javax.inject.Inject

class AppControlActivityTest : NannyAppTestCase() {

    private lateinit var controller: ActivityController<AppControlActivity>
    private lateinit var appControlActivity: AppControlActivity
    @Inject lateinit var appControlBinder: AppControlBinder
    @Mock internal lateinit var menu: Menu
    @Mock internal lateinit var menuItem: MenuItem

    @Before
    fun setUp() {
        controller = Robolectric.buildActivity(AppControlActivity::class.java)
        appControlActivity = controller.get()
        val contextComponent = MockComponentFactory.activityComponent
        appControlActivity.setTestComponent(contextComponent)
        contextComponent.inject(this)
        controller.setup()
    }

    @Test
    fun lifecycle() {
        controller.pause().stop().destroy()
        verify(appControlBinder).onCreate(null)
        verify(appControlBinder).onResume()
        verify(appControlBinder).onPause()
    }

    @Test
    fun onCreateOptionsMenu() {
        appControlActivity.onCreateOptionsMenu(menu)

        verify(appControlBinder).onCreateOptionsMenu(menu)
    }

    @Test
    fun onOptionsItemSelected() {
        appControlActivity.onOptionsItemSelected(menuItem)

        verify(appControlBinder).onOptionsItemSelected(menuItem)
    }
}
